docs/extensions/extension-registry.md

```diff
@@ -0,0 +1,580 @@
+          - name: artifact-store
+            service_resource: s3
+    ```
+
+4. this workflow step explicitly uses FuseML workflow expressions to map new environment variable names to the configuration entries from the extension record instead of relying on the default ones extracted from the extension record. It also accesses other fields in the extension record to perform additional actions:
+
+    ```yaml
+    [...]
+    steps:
+      [...]
+      - name: trainer
+        [...]
+        extensions:
+          - name: experiment-tracking
+            product: mlflow
+            service_resource: mlflow-tracking
+          - name: artifact-store
+            product: mlflow
+            service_resource: s3
+        env:
+          - name: S3_ACCESS_KEY_ID
+            value: '{{ extension.artifact-store.cfg.AWS_ACCESS_KEY_ID }}'
+          - name: S3_SECRET_ACCESS_KEY
+            value: '{{ extension.artifact-store.cfg.AWS_SECRET_ACCESS_KEY }}'
+          - name: MLFLOW_TRACKER_URL
+            value: '{{ extension.experiment-tracking.url }}'
+          - name: MLFLOW_TRACKER_VERSION
+            value: '{{ extension.experiment-tracking.version }}'
+          - name: MLFLOW_TRACKER_ZONE
+            value: '{{ extension.experiment-tracking.zone }}'
+    ```
+
+5. a similar approach is to use the FuseML workflow expressions to expand fields in the extension record and use them as values for workflow step inputs:
+
+    ```yaml
+    [...]
+    steps:
+      [...]
+      - name: trainer
+        [...]
+        inputs:
+          [...]
+          - name: s3_access_key
+            value: '{{ extension.artifact-store.cfg.AWS_ACCESS_KEY_ID }}'
+          - name: s3_secret_access_key
+            value: '{{ extension.artifact-store.cfg.AWS_SECRET_ACCESS_KEY }}'
+          - name: s3_endpoint_url
+            value: '{{ extension.artifact-store.url }}'
+        extensions:
+          - name: artifact-store
+            service_resource: s3
+    ```
+
+## Extension Registry Use Cases
+
+The extension registry targets some of the following use-cases:
+
+1. as an Ops engineer (MLOps/DevOps/ITOps), I need a way to configure my FuseML instance with the parameters required to dynamically integrate with 3rd party AI/ML tools (e.g. URLs, endpoints, credentials, other tool specific configuration attributes). For non-production environments, this requirement can also come from ML Engineers or even Data Scientists that are looking to quickly set up FuseML for experimentation purposes.
+
+    Examples:
+
+    * DS: I have an MLFlow tracking service already set up by me that I use for my ML experiments and I want to reuse it for FuseML automated workflows. I will configure FuseML with the information it needs to access the MLFlow tracking service (the tracker service URL and the hostname, username and keys for the storage backend). The information is stored in a central registry where FuseML workflows can access it.
```

Comment:
I think it would be better to use full "Data Scientist" role name here

Reply:
(it's actually bit strange that the use case 1 is started as a use case for  Ops engineer, but then you proceed with 2 examples for DS ...

Reply:
Done. I also expanded "Ops" into "Ops engineer"